# 4.6: Combining & Exporting Data

### This script contains the following points:

#### 1. Import the data sets into Jupyter

#### 2. Check the dimensions of the imported dataframes

#### 3. Merge the dataframes

#### 4. Use the indicator argument to check whether there was a full match between the two dataframes

#### 5. Export data in pickle format

In [8]:
# Import libraries
import pandas as pd
import numpy as np
import os

### 1. Import the data sets into Jupyter

In [11]:
path = r'C:\Users\IVON6\Instacart Basket Analysis'

In [13]:
path

'C:\\Users\\IVON6\\Instacart Basket Analysis'

In [15]:
#products data
df_prods = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'products_cleanData.csv'), index_col = False)

#orders data
df_ords = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'orders_cleanData.csv'), index_col = False)

# Import dataset orders_products_prior
df_ords_prior = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders_products_prior.csv'), index_col = False)

### 2. Check the dimension of the imported dataframes

In [18]:
# Check the output
df_ords_prior.head()

order_id  product_id  add_to_cart_order  reordered
0         2       33120                  1          1
1         2       28985                  2          1
2         2        9327                  3          0
3         2       45918                  4          1
4         2       30035                  5          0

In [20]:
df_ords.head()

Unnamed: 0  order_id  user_id  order_number  orders_day_of_week  \
0           0   2539329        1             1                   2   
1           1   2398795        1             2                   3   
2           2    473747        1             3                   3   
3           3   2254736        1             4                   4   
4           4    431534        1             5                   4   

   order_hour_of_day  days_since_prior_order  first_order  
0                  8                     NaN         True  
1                  7                    15.0        False  
2                 12                    21.0        False  
3                  7                    29.0        False  
4                 15                    28.0        False

In [22]:
# Method 1: Delete Unnamed Columns
df_ords.drop('Unnamed: 0', axis=1, inplace=True)

In [24]:
df_ords.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2398795        1             2                   3                  7   
2    473747        1             3                   3                 12   
3   2254736        1             4                   4                  7   
4    431534        1             5                   4                 15   

   days_since_prior_order  first_order  
0                     NaN         True  
1                    15.0        False  
2                    21.0        False  
3                    29.0        False  
4                    28.0        False

In [28]:
df_ords.shape

(3421083, 7)

In [30]:
df_ords_prior.shape

(32434489, 4)

### 3. Merge the dataframes

In [33]:
df_merged_large = df_ords.merge(df_ords_prior, on = 'order_id', indicator = True)

In [35]:
df_merged_large.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2539329        1             1                   2                  8   
2   2539329        1             1                   2                  8   
3   2539329        1             1                   2                  8   
4   2539329        1             1                   2                  8   

   days_since_prior_order  first_order  product_id  add_to_cart_order  \
0                     NaN         True         196                  1   
1                     NaN         True       14084                  2   
2                     NaN         True       12427                  3   
3                     NaN         True       26088                  4   
4                     NaN         True       26405                  5   

   reordered _merge  
0          0   both  
1          0   both  
2          0   both  
3          0   both  
4          0   both

### 4. Use the indicator argument to check whether there was a full match between the two dataframes

In [38]:
df_merged_large['_merge'].value_counts()

_merge
both          32434489
left_only            0
right_only           0
Name: count, dtype: int64

#### The dataframe after merging has 32,434,489 rows, and each of those rows have information found in both input data sets.

### 5. Export data in pickle format

In [43]:
# Export data to pkl
df_merged_large.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'orders_products_combined.pkl'))

In [ ]:
# Export data to csv
df_merged_large.to_csv(os.path.join(path, '02 Data','Prepared Data', 'orders_products_combined.csv'))